path: compile error on Fedora 33 ASAN GCC 10.2.1

## Steps to Reproduce the Problem

Build Elektra on on Fedora 33 with ASAN using GCC 10.2.1.

## Actual Result

```
/home/jenkins/workspace/libelektra_PR-3532/src/plugins/path/path.c: In function 'validateKey':
/home/jenkins/workspace/libelektra_PR-3532/src/plugins/path/path.c:72:7: error: argument 2 is null but the corresponding size argument 3 value is 1000 [-Werror=nonnull]
   72 |   if (strerror_r (errno, errmsg, ERRORMSG_LENGTH) != 0)
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/jenkins/workspace/libelektra_PR-3532/src/include/kdbplugin.h:16,
                 from /home/jenkins/workspace/libelektra_PR-3532/src/plugins/path/path.h:15,
                 from /home/jenkins/workspace/libelektra_PR-3532/src/plugins/path/path.c:9:
/usr/include/string.h:428:14: note: in a call to function 'strerror_r' declared with attribute 'write_only (2, 3)'
  428 | extern char *strerror_r (int __errnum, char *__buf, size_t __buflen)
      |              ^~~~~~~~~~
```

Error discovered in #3532.

## System Information

- Elektra Version: master

## Further Log Files and Output
[log_path.txt](https://github.com/ElektraInitiative/libelektra/files/5504505/log_path.txt)
[Jenkins CI run](https://build.libelektra.org/blue/organizations/jenkins/libelektra/detail/PR-3532/4/pipeline/678)

--- a/doc/news/_preparation_next_release.md
+++ b/doc/news/_preparation_next_release.md
@@ -173,6 +173,8 @@ you up to date with the multi-language support provided by Elektra.
 - We now use Debian sid to build the documentation instead of Debian stretch. The Doxygen version in Debian stretch [contains a bug](https://github.com/doxygen/doxygen/issues/6456) that causes the generation of the PDF documentation to fail. _(René Schwaiger)_
 - <<TODO>>
 - <<TODO>>
+- Temporarily resolve cyclic dependency between `go-elektra` and `libelektra` builds. _(Mihael Pranjić)_
+- Add a new Jenkinsfile for release automation. _(Robert Sowula)_
 
 ### Travis
 

--- a/doc/news/_preparation_next_release.md
+++ b/doc/news/_preparation_next_release.md
@@ -174,10 +174,9 @@ you up to date with the multi-language support provided by Elektra.
 
 - We now use Debian sid to build the documentation instead of Debian stretch. The Doxygen version in Debian stretch [contains a bug](https://github.com/doxygen/doxygen/issues/6456) that causes the generation of the PDF documentation to fail. _(René Schwaiger)_
 - Use Fedora 33 and 32, drop Fedora 31 use in Jenkins. _(Mihael Pranjić)_
+- Use Fedora 33 and 32, drop Fedora 31 use in Jenkins. _(Mihael Pranjić)_
 - <<TODO>>
 - <<TODO>>
-- Temporarily resolve cyclic dependency between `go-elektra` and `libelektra` builds. _(Mihael Pranjić)_
-- Add a new Jenkinsfile for release automation. _(Robert Sowula)_
 
 ### Travis
 

--- a/doc/news/_preparation_next_release.md
+++ b/doc/news/_preparation_next_release.md
@@ -176,6 +176,8 @@ you up to date with the multi-language support provided by Elektra.
 - Use Fedora 33 and 32, drop Fedora 31 use in Jenkins. _(Mihael Pranjić)_
 - <<TODO>>
 - <<TODO>>
+- Temporarily resolve cyclic dependency between `go-elektra` and `libelektra` builds. _(Mihael Pranjić)_
+- Add a new Jenkinsfile for release automation. _(Robert Sowula)_
 
 ### Travis
 

--- a/doc/news/_preparation_next_release.md
+++ b/doc/news/_preparation_next_release.md
@@ -174,7 +174,6 @@ you up to date with the multi-language support provided by Elektra.
 
 - We now use Debian sid to build the documentation instead of Debian stretch. The Doxygen version in Debian stretch [contains a bug](https://github.com/doxygen/doxygen/issues/6456) that causes the generation of the PDF documentation to fail. _(René Schwaiger)_
 - Use Fedora 33 and 32, drop Fedora 31 use in Jenkins. _(Mihael Pranjić)_
-- Use Fedora 33 and 32, drop Fedora 31 use in Jenkins. _(Mihael Pranjić)_
 - <<TODO>>
 - <<TODO>>
 

--- a/doc/news/_preparation_next_release.md
+++ b/doc/news/_preparation_next_release.md
@@ -158,23 +158,24 @@ you up to date with the multi-language support provided by Elektra.
 
 - We added a Docker image for [building the documentation on Debian sid](../../scripts/docker/debian/sid/doc.Dockerfile). _(René Schwaiger)_
 - We removed the Docker image for building the documentation on Debian Stretch. _(René Schwaiger)_
+- Add Fedora 33 Dockerfile for Cirrus and Jenkins CI. _(Mihael Pranjić)_
+- <<TODO>>
 - <<TODO>>
 
 ## Infrastructure
 
 ### Cirrus
 
-- <<TODO>>
+- Upgrade Cirrus Fedora docker image to Fedora 33. _(Mihael Pranjić)_
 - <<TODO>>
 - <<TODO>>
 
 ### Jenkins
 
 - We now use Debian sid to build the documentation instead of Debian stretch. The Doxygen version in Debian stretch [contains a bug](https://github.com/doxygen/doxygen/issues/6456) that causes the generation of the PDF documentation to fail. _(René Schwaiger)_
+- Use Fedora 33 and 32, drop Fedora 31 use in Jenkins. _(Mihael Pranjić)_
 - <<TODO>>
 - <<TODO>>
-- Temporarily resolve cyclic dependency between `go-elektra` and `libelektra` builds. _(Mihael Pranjić)_
-- Add a new Jenkinsfile for release automation. _(Robert Sowula)_
 
 ### Travis
 

--- a/scripts/docker/cirrus/fedora/Dockerfile
+++ b/scripts/docker/cirrus/fedora/Dockerfile
@@ -1,6 +1,6 @@
-FROM fedora:32
+FROM fedora:33
 
-RUN dnf upgrade -y && dnf install -y \
+RUN dnf upgrade --refresh -y && dnf install -y \
         augeas-devel \
         bison \
         boost-devel \
@@ -31,7 +31,7 @@ RUN dnf upgrade -y && dnf install -y \
         openssl-devel \
         procps-ng \
         python3-devel \
-        qt5-devel \
+        qt5-qtbase-devel \
         ruby-devel \
         rubygem-test-unit \
         swig \
@@ -51,3 +51,4 @@ RUN mkdir -p ${GTEST_ROOT} \
       -L https://github.com/google/googletest/archive/${GTEST_VER}.tar.gz \
     && tar -zxvf gtest.tar.gz --strip-components=1 -C ${GTEST_ROOT} \
     && rm gtest.tar.gz
+

--- a/scripts/docker/cirrus/fedora/Dockerfile
+++ b/scripts/docker/cirrus/fedora/Dockerfile
@@ -3,6 +3,7 @@ FROM fedora:33
 RUN dnf upgrade --refresh -y && dnf install -y \
         augeas-devel \
         bison \
+        bison-devel \
         boost-devel \
         cmake \
         curl \
@@ -11,19 +12,23 @@ RUN dnf upgrade --refresh -y && dnf install -y \
         file \
         findutils \
         flex \
+        flex-devel \
         gcc-c++ \
         git \
         glib2 \
         gpgme-devel \
         java-1.8.0-openjdk-devel \
         jna \
+        libasan \
         libcurl-devel \
         libev-devel \
         libgcrypt-devel \
         libgit2-devel \
         libmarkdown-devel \
+        libubsan \
         libuv-devel \
         libxml2-devel \
+        llvm \
         lua-devel \
         make \
         maven \

--- a/scripts/docker/fedora/33/Dockerfile
+++ b/scripts/docker/fedora/33/Dockerfile
@@ -54,6 +54,7 @@ RUN dnf upgrade --refresh -y && dnf install -y \
         glibc \
         gpgme \
         libgcrypt \
+        ruby-libs \
     && dnf clean all -y
 
 # Google Test

--- a/scripts/docker/fedora/33/Dockerfile
+++ b/scripts/docker/fedora/33/Dockerfile
@@ -1,6 +1,6 @@
 FROM fedora:33
 
-RUN dnf upgrade -y && dnf install -y \
+RUN dnf upgrade --refresh -y && dnf install -y \
         augeas-devel \
         bison \
         bison-devel \
@@ -37,7 +37,7 @@ RUN dnf upgrade -y && dnf install -y \
         openssl-devel \
         procps-ng \
         python3-devel \
-        qt5-devel \
+        qt5-qtbase-devel \
         ruby-devel \
         rubygem-test-unit \
         swig \

--- a/scripts/docker/fedora/33/Dockerfile
+++ b/scripts/docker/fedora/33/Dockerfile
@@ -1,0 +1,89 @@
+FROM fedora:33
+
+RUN dnf upgrade -y && dnf install -y \
+        augeas-devel \
+        bison \
+        bison-devel \
+        boost-devel \
+        cmake \
+        curl \
+        dbus-devel \
+        diffutils \
+        dnf-plugins-core \
+        file \
+        findutils \
+        flex \
+        flex-devel \
+        gcc-c++ \
+        git \
+        glib2 \
+        gpgme-devel \
+        java-1.8.0-openjdk-devel \
+        jna \
+        libasan \
+        libcurl-devel \
+        libev-devel \
+        libgcrypt-devel \
+        libgit2-devel \
+        libmarkdown-devel \
+        libubsan \
+        libuv-devel \
+        libxml2-devel \
+        llvm \
+        lua-devel \
+        make \
+        maven \
+        ninja-build \
+        openssl-devel \
+        procps-ng \
+        python3-devel \
+        qt5-devel \
+        ruby-devel \
+        rubygem-test-unit \
+        swig \
+        valgrind \
+        xerces-c-devel \
+        yajl-devel \
+        yaml-cpp-devel \
+        zlib-devel \
+    && dnf debuginfo-install -y \
+        bison \
+        boost \
+        flex \
+        glib2 \
+        glibc \
+        gpgme \
+        libgcrypt \
+    && dnf clean all -y
+
+# Google Test
+ENV GTEST_ROOT=/opt/gtest
+ARG GTEST_VER=release-1.10.0
+RUN mkdir -p ${GTEST_ROOT} \
+    && cd /tmp \
+    && curl -o gtest.tar.gz \
+      -L https://github.com/google/googletest/archive/${GTEST_VER}.tar.gz \
+    && tar -zxvf gtest.tar.gz --strip-components=1 -C ${GTEST_ROOT} \
+    && rm gtest.tar.gz
+
+# Create User:Group
+# The id is important as jenkins docker agents use the same id that is running
+# on the slaves to execute containers
+ARG JENKINS_GROUPID
+RUN groupadd \
+    -g ${JENKINS_GROUPID} \
+    -f \
+    jenkins
+
+ARG JENKINS_USERID
+RUN useradd \
+    --create-home \
+    --uid ${JENKINS_USERID} \
+    --gid ${JENKINS_GROUPID} \
+    --shell "/bin/bash" \
+    jenkins
+USER ${JENKINS_USERID}
+
+RUN git config --global user.email 'Jenkins <autobuilder@libelektra.org>' \
+    && git config --global user.name 'Jenkins'
+

--- a/scripts/jenkins/Jenkinsfile
+++ b/scripts/jenkins/Jenkinsfile
@@ -326,18 +326,18 @@ def dockerInit() {
       "./scripts/docker/ubuntu/bionic/Dockerfile"
     )
 
-    /* Fedora 31 image*/
-    DOCKER_IMAGES.fedora_31 = createDockerImageDesc(
-      "fedora-31", this.&idTesting,
-      "./scripts/docker/fedora/31",
-      "./scripts/docker/fedora/31/Dockerfile"
-    )
-
     /* Fedora 32 image*/
     DOCKER_IMAGES.fedora_32 = createDockerImageDesc(
       "fedora-32", this.&idTesting,
       "./scripts/docker/fedora/32",
       "./scripts/docker/fedora/32/Dockerfile"
+    )
+
+    /* Fedora 33 image*/
+    DOCKER_IMAGES.fedora_33 = createDockerImageDesc(
+      "fedora-33", this.&idTesting,
+      "./scripts/docker/fedora/33",
+      "./scripts/docker/fedora/33/Dockerfile"
     )
 
     /* Image building the libelektra.org website */
@@ -466,8 +466,8 @@ def generateMainBuildStages() {
   tasks.failFast = true
 
   tasks << buildAndTest(
-    "fedora-32-full",
-    DOCKER_IMAGES.fedora_32,
+    "fedora-33-full",
+    DOCKER_IMAGES.fedora_33,
     CMAKE_FLAGS_BUILD_ALL +
       CMAKE_FLAGS_DEBUG +
       CMAKE_FLAGS_BUILD_FULL +
@@ -513,8 +513,8 @@ def generateFullBuildStages() {
   tasks << buildDoc()
 
   tasks << buildAndTest(
-    "fedora-31-full",
-    DOCKER_IMAGES.fedora_31,
+    "fedora-32-full",
+    DOCKER_IMAGES.fedora_32,
     CMAKE_FLAGS_BUILD_ALL +
       CMAKE_FLAGS_DEBUG +
       CMAKE_FLAGS_BUILD_FULL +
@@ -525,8 +525,8 @@ def generateFullBuildStages() {
   // Build Elektra with clang and ASAN
   // Detects memory leaks via ASAN
   tasks << buildAndTestAsan(
-    "fedora-32-asan",
-    DOCKER_IMAGES.fedora_32,
+    "fedora-33-asan",
+    DOCKER_IMAGES.fedora_33,
     CMAKE_FLAGS_BUILD_ALL
   )
 

--- a/src/bindings/swig/lua/README.md
+++ b/src/bindings/swig/lua/README.md
@@ -1,4 +1,4 @@
-- infos = Information about elektrify getenv below
+- infos = Information about Lua bindings below
 - infos/author = Manuel Mausz <manuel-elektra@mausz.at>
 - infos/status = maintained
 - infos/provides = swig

--- a/src/bindings/swig/python/README.md
+++ b/src/bindings/swig/python/README.md
@@ -1,4 +1,4 @@
-- infos =
+- infos = Information about Python bindings below
 - infos/author = Manuel Mausz <manuel-elektra@mausz.at>
 - infos/status = maintained
 - infos/provides = swig

--- a/src/plugins/path/path.c
+++ b/src/plugins/path/path.c
@@ -69,6 +69,7 @@ static int validateKey (Key * key, Key * parentKey)
 	{
 		char * errmsg = elektraMalloc (ERRORMSG_LENGTH + 1 + keyGetNameSize (key) + keyGetValueSize (key) +
 					       sizeof ("name:  value:  message: "));
+		if (!errmsg) return -1;
 		if (strerror_r (errno, errmsg, ERRORMSG_LENGTH) != 0)
 		{
 			strcpy (errmsg, "Unknown error");

